chain/client/src/client_actor.rs

```diff
@@ -1784,6 +1782,37 @@ impl ClientActor {
             &self.config_updater,
         )
     }
+
+    /// Checks if the node is syncing its State and applies special logic in that case.
+    /// A node usually ignores blocks that are too far ahead, but in case of a node syncing its state it is looking for 2 specific blocks:
+    /// * The first block of the new epoch
+    /// * The last block of the prev epoch
+    /// Returns whether the node is syncing its state.
+    fn maybe_receive_state_sync_blocks(&mut self, block: &Block) -> bool {
+        match self.client.sync_status {
+            SyncStatus::StateSync(StateSyncStatus { sync_hash, .. }) => {
```

Comment:
nit: maybe this could be an if? or better yet 
```
let SyncStatus::StateSync(StateSyncStatus { sync_hash, .. }) = self.client.sync_status else { return false; } 
```